tee-worker/omni-executor/native-task-handler/src/lib.rs

```diff
@@ -298,51 +298,51 @@ async fn handle_native_task<
 				return;
 			}
 
-			let call = OmniAccountCall::request_intent { intent: intent.to_subxt_type() };
-			let dispatch_as_omni_account_call =
-				parentchain_api_interface::tx().omni_account().dispatch_as_omni_account(
-					sender.hash().to_subxt_type(),
-					RuntimeCall::OmniAccount(call),
-					auth_type.clone().map(|t| t.to_subxt_type()),
-				);
-
-			let tx = ctx.transaction_signer.sign(dispatch_as_omni_account_call).await;
-			if rpc_client.submit_tx(&tx).await.is_err() {
-				send_error(
-					"Failed to submit RequestIntent tx".to_string(),
-					response_sender,
-					NativeTaskError::InternalError,
-				);
-				ctx.transaction_signer.update_nonce().await;
-				return;
-			}
-
-			let mut execution_result = IntentCompletedDetail::Success;
 			let omni_account = sender.to_omni_account();
-
-			let tx = match intent {
+			let _ = notify_intent_accepted(
+				&mut rpc_client,
+				ctx.transaction_signer.clone(),
+				omni_account.clone(),
+				intent_id,
+				intent.clone(),
+			)
+			.await;
+
+			let (execution_result, should_notify_parentchain) = match intent {
 				Intent::SystemRemark(remark) => {
 					let remark_call = SystemCall::remark { remark: remark.to_vec() };
-					let dispatch_as_omni_account_call =
-						parentchain_api_interface::tx().omni_account().dispatch_as_signed(
-							sender.hash().to_subxt_type(),
-							RuntimeCall::System(remark_call),
-							auth_type.map(|t| t.to_subxt_type()),
-						);
-					ctx.transaction_signer.sign(dispatch_as_omni_account_call).await
+					let _ = dispatch_as_omni_account(
+						&mut rpc_client,
+						ctx.transaction_signer.clone(),
+						sender,
+						RuntimeCall::System(remark_call),
+						auth_type,
+					)
+					.await;
+					send_ok(
+						response_sender,
+						NativeTaskOk::RequestIntentResult { intent_id, success: true },
+					);
+					(IntentCompletedDetail::Success, true)
 				},
 				Intent::TransferNative(transfer) => {
 					let transfer_call = BalancesCall::transfer_allow_death {
 						dest: transfer.to.to_subxt_type().into(),
 						value: transfer.value,
 					};
-					let dispatch_as_omni_account_call =
-						parentchain_api_interface::tx().omni_account().dispatch_as_signed(
-							sender.hash().to_subxt_type(),
-							RuntimeCall::Balances(transfer_call),
-							auth_type.map(|t| t.to_subxt_type()),
-						);
-					ctx.transaction_signer.sign(dispatch_as_omni_account_call).await
+					let _ = dispatch_as_omni_account(
```

Comment:
This call should use `dispatch_as_signed`.  `SystemCall::remark` also needs to be changed